Source/Urho3D/Resource/GraphNode.h

```diff
@@ -0,0 +1,198 @@
+
+    /// Return ID.
+    /// @property{get_id}
+    unsigned GetID() const { return id_; }
+
+    /// Return Graph.
+    /// @property{get_graph}
+    Graph* GetGraph() const { return graph_; }
+
+    /// Return name.
+    /// @property
+    const ea::string& GetName() const { return name_; }
+
+    /// Return name hash.
+    StringHash GetNameHash() const { return nameHash_; }
+
+    /// Get or add node property.
+    Variant& GetOrAddProperty(const ea::string_view name);
+
+    /// Get node property. Returns nullptr if property is not found.
+    Variant* GetProperty(const ea::string_view name);
+
+    /// Get properties.
+    ea::span<GraphNodeProperty> GetProperties() { return ea::span(properties_); }
+
+    /// Add property with value.
+    GraphNode* WithProperty(const ea::string_view name, const Variant& value);
+
+    /// Get input pin by name.
+    GraphDataInPin* GetInput(const ea::string_view name);
+
+    /// Get input pins.
+    ea::span<GraphDataInPin> GetInputs() { return ea::span(inputPins_); }
+
+    /// Get or add input pin.
+    GraphDataInPin& GetOrAddInput(const ea::string_view name);
+
+    /// Add input pin.
+    GraphNode* WithInput(const ea::string_view name, VariantType type = VAR_NONE);
+
+    /// Add input pin with default value.
+    GraphNode* WithInput(const ea::string_view name, const Variant& value);
+
+    /// Add input pin connected to the output pin.
+    GraphNode* WithInput(const ea::string_view name, GraphOutPin* pin, VariantType type = VAR_NONE);
+
+    /// Get or add output pin.
+    GraphOutPin& GetOrAddOutput(const ea::string_view name);
+
+    /// Add output pin.
+    GraphNode* WithOutput(const ea::string_view name, VariantType type = VAR_NONE);
+
+    /// Get output pin.
+    GraphOutPin* GetOutput(const ea::string_view name);
+
+    /// Get output pins.
+    ea::span<GraphOutPin> GetOutputs() { return ea::span(outputPins_); }
+
+    /// Get or add enter pin.
+    GraphOutPin& GetOrAddEnter(const ea::string_view name);
+
+    /// Get enter pin.
+    GraphOutPin* GetEnter(const ea::string_view name);
+
+    /// Get output pins.
+    ea::span<GraphOutPin> GetEnters() { return ea::span(enterPins_); }
+
+    /// Get exit pin.
+    GraphInPin* GetExit(const ea::string_view name);
+
+    /// Get or add exit pin.
+    GraphInPin& GetOrAddExit(const ea::string_view name);
+
+    /// Get exit pins.
+    ea::span<GraphInPin> GetExits() { return ea::span(exitPins_); }
+
+    /// Set name of the graph node.
+    /// @property
+    void SetName(const ea::string& name);
+
+    /// Serialize from/to archive. Return true if successful.
+    bool Serialize(Archive& archive);
+
+protected:
+
+    /// Set graph and id. Called by Graph.
+    void SetGraph(Graph* scene, unsigned id);
+
+    template <typename Iterator> bool SerializePins(Archive& archive, Iterator begin, Iterator end);
+
+private:
+    /// Name.
+    ea::string name_;
+    /// Name hash.
+    StringHash nameHash_;
+
+    /// Graph.
+    Graph* graph_;
```

Comment:
Please never leave unitilialized variables and members, even if you have a ctor

Reply:
Again, I was following Urho3D approach.